tests/unit-tests/test_dataloading.py

```diff
@@ -1354,6 +1355,92 @@ def test_inbatch_joint_neg_sampler(num_pos, num_neg):
         assert_equal(in_batch_dst[i*(num_pos-1):(i+1)*(num_pos-1)].numpy(),
                      np.arange(num_pos)[tmp_idx])
 
+def test_GSgnnTrainData_homogeneous():
+    # initialize the torch distributed environment
+    th.distributed.init_process_group(backend='gloo',
+                                      init_method='tcp://127.0.0.1:23456',
+                                      rank=0,
+                                      world_size=1)
+    tr_ntypes = ["_N"]
+    va_ntypes = ["_N"]
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        dist_graph, part_config = generate_dummy_dist_graph(graph_name='dummy',
```

Comment:
Is the dummpy dist graph a heterogeneous graph?
Can you add some comments here how the exceptions are triggerred?

Reply:
Refer to the comments here about what each test case is and why it should trigger an error.